From cd24c9fbba72b179e32c560ea3b302f1e8bab9bb Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Wed, 2 Apr 2025 16:56:33 +0800
Subject: [PATCH 080/123] mtk: mt76: mt7996: support EDCCA standard ETSI_2023

Support the new EDCCA standard, ETSI_2023, as the original ETSI standard
is deprecated.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 mt7996/mtk_mcu.c | 11 +++--------
 mt7996/mtk_mcu.h |  3 ++-
 2 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/mt7996/mtk_mcu.c b/mt7996/mtk_mcu.c
index 5133afa3..5679c9e5 100644
--- a/mt7996/mtk_mcu.c
+++ b/mt7996/mtk_mcu.c
@@ -62,9 +62,6 @@ int mt7996_mcu_muru_dbg_info(struct mt7996_dev *dev, u16 item, u8 val)
 
 int mt7996_mcu_edcca_enable(struct mt7996_phy *phy, bool enable)
 {
-	struct mt7996_dev *dev = phy->dev;
-	struct cfg80211_chan_def *chandef = &phy->mt76->chandef;
-	enum nl80211_band band = chandef->chan->band;
 	struct {
 		u8 band_idx;
 		u8 _rsv[3];
@@ -82,17 +79,15 @@ int mt7996_mcu_edcca_enable(struct mt7996_phy *phy, bool enable)
 		.std = EDCCA_DEFAULT,
 	};
 
-	switch (dev->mt76.region) {
+	switch (phy->dev->mt76.region) {
 	case NL80211_DFS_JP:
 		req.std = EDCCA_JAPAN;
 		break;
 	case NL80211_DFS_FCC:
-		if (band == NL80211_BAND_6GHZ)
-			req.std = EDCCA_FCC;
+		req.std = EDCCA_FCC;
 		break;
 	case NL80211_DFS_ETSI:
-		if (band == NL80211_BAND_6GHZ)
-			req.std = EDCCA_ETSI;
+		req.std = EDCCA_ETSI_2023;
 		break;
 	default:
 		break;
diff --git a/mt7996/mtk_mcu.h b/mt7996/mtk_mcu.h
index 988f13a5..085aa641 100644
--- a/mt7996/mtk_mcu.h
+++ b/mt7996/mtk_mcu.h
@@ -118,7 +118,8 @@ enum {
 	EDCCA_DEFAULT = 0,
 	EDCCA_FCC = 1,
 	EDCCA_ETSI = 2,
-	EDCCA_JAPAN = 3
+	EDCCA_JAPAN = 3,
+	EDCCA_ETSI_2023 = 4,
 };
 
 enum {
-- 
2.45.2

